libvast/src/logical_pipeline.cpp

```diff
@@ -327,17 +373,17 @@ auto logical_pipeline::make(std::vector<logical_operator_ptr> ops)
     return caf::make_error(ec::invalid_argument,
                            fmt::format("invalid operator at index {}",
                                        std::distance(ops.begin(), invalid)));
-  const auto mismatch
+  const auto violation
     = std::adjacent_find(ops.begin(), ops.end(), [](auto& a, auto& b) {
         return a->output_element_type() != b->input_element_type()
-               && a->output_element_type().id != element_type_id<void>;
+               || a->output_element_type().id == element_type_id<void>;
       });
-  if (mismatch != ops.end()) {
+  if (violation != ops.end()) {
     return caf::make_error(
       ec::invalid_argument,
-      fmt::format("element type mismatch: cannot connect {} -> {}",
-                  (*mismatch)->output_element_type().name,
-                  (*(mismatch + 1))->input_element_type().name));
+      fmt::format("pipeline type violation: cannot connect {} -> {}",
+                  (*violation)->output_element_type().name,
+                  (*(violation + 1))->input_element_type().name));
```

Comment:
If the pipeline consists of a single operator than this derefenceses past the end of the vector of operators.